taichi/backends/vulkan/aot_module_builder_impl.h

```diff
@@ -0,0 +1,52 @@
+#pragma once
+
+#include <string>
+#include <vector>
+
+#include "taichi/backends/vulkan/aot_utils.h"
+#include "taichi/backends/vulkan/snode_struct_compiler.h"
+#include "taichi/backends/vulkan/runtime.h"
+#include "taichi/backends/vulkan/kernel_utils.h"
+
+#include "taichi/program/aot_module_builder.h"
+
+namespace taichi {
+namespace lang {
+namespace vulkan {
+
+class AotModuleBuilderImpl : public AotModuleBuilder {
+ public:
+  explicit AotModuleBuilderImpl(
+      VkRuntime *runtime,
+      const std::vector<CompiledSNodeStructs> &compiled_structs);
+  void dump(const std::string &output_dir,
+            const std::string &filename) const override;
+
+ protected:
```

Comment:
nit: I belive this class is not gonna be further inherited.
```suggestion
 private:
```

Reply:
I agree, I moved it to private and if it needs to be inherited and used later... we can still change it back to protected :).